addons/mrp_subcontracting_purchase/models/stock_valuation_layer.py

```diff
@@ -17,4 +17,4 @@ def _get_layer_price_unit(self):
             # we need to get the components used for each quantity
             for move in production.move_raw_ids:
                 components_price += sum(move.sudo().stock_valuation_layer_ids.mapped('value')) / production.product_uom_qty
-        return super()._get_layer_price_unit() - components_price
+        return super()._get_layer_price_unit() + components_price
```

Comment:
```suggestion
        return super()._get_layer_price_unit() - abs(components_price)
```
May be a bit more obvious on what we're trying to achieve. Also may be a tad more safe in case we somehone end up with a positive `components_price` for whatever reason? :thinking:

Reply:
what about changing the += in the line above to a -=

Reply:
not sure it's the best approach to force the value to be positive 
I don't have all use cases in mind but if the component itself is subcontracted this could lead to an issue then

Reply:
> I don't have all use cases in mind but if the component itself is subcontracted this could lead to an issue then

Yeah but in that case, adding the positive value would be wrong as well, wouldn't it?
Your change switches the line to an addition based on the fact that `components_price` *should be* negative.